sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -115,35 +117,17 @@ export class InstitutionLocationInstitutionsController extends BaseController {
    * Update an institution location.
    * @param locationId
    * @param payload
-   * @returns number of updated rows.
+   * @returns InstitutionLocation.
    */
-  @ApiBadRequestResponse({
-    description: "Invalid request to update the institution location.",
-  })
   @HasLocationAccess("locationId")
   @IsInstitutionAdmin()
   @Patch(":locationId")
   async update(
     @Param("locationId") locationId: number,
-    @Body() payload: InstitutionLocationFormAPIInDTO,
-    @UserToken() userToken: IInstitutionUserToken,
-  ): Promise<void> {
-    // Validate the location data that will be saved to SIMS DB.
-    const dryRunSubmissionResult = await this.formService.dryRunSubmission(
-      FormNames.InstitutionLocation,
+    @Body() payload: InstitutionLocationPrimaryContactAPIInDTO,
+  ): Promise<InstitutionLocation> {
+    return this.locationService.updateLocationPrimaryContact(
```

Comment:
Should this method have such a specific name `updateLocationPrimaryContact` (same for DTO)?
This endpoint is to allow the institution to update the data (that right now is just the primary contact), so, if in the future we need to update any other column the name will no longer be valid.